### PR TITLE
Hierarchy Restructure + Multiple Face Masks + Multiple Face Meshes + Handling Differerent Plugin Versions 

### PR DESCRIPTION
**Pull request by gazuntype**
yeah, this enables all of those functionality!

### CODE REVIEW

File: Simulator/Simulator.cs [hunk 1]
```diff
@@ -100,19 +169,28 @@ public class Simulator : MonoBehaviour
...
         faceTracker.localPosition = faceData.face.localPosition;
         faceTracker.localEulerAngles = faceData.face.localRotation;
-        leftEyeTracker.localPosition = faceData.leftEye.position;
-        leftEyeTracker.localEulerAngles = faceData.leftEye.rotation;
-        rightEyeTracker.localPosition = faceData.rightEye.position;
-        rightEyeTracker.localEulerAngles = faceData.rightEye.rotation;
+        leftEyeTracker.localPosition = faceData.leftEye.localPosition;
+        leftEyeTracker.localEulerAngles = faceData.leftEye.localRotation;
+        rightEyeTracker.localPosition = faceData.rightEye.localPosition;
+        rightEyeTracker.localEulerAngles = faceData.rightEye.localRotation;
```
Comment (chornesays) [inline Simulator/Simulator.cs:175]: What about the situation where the trackers are null? Artists may delete these objects.

File: Simulator/Simulator.cs [hunk 2]
```diff
@@ -334,5 +455,40 @@ public struct FaceRecording
...
+                if (GUILayout.Button("Stop")){
+                    sim.isPlaying = false;
+                }
+            }
+            else{
+                if (GUILayout.Button("Play")){
+                    sim.isPlaying = true;
+                }
+            }
+            if (!sim.IsSetUpProperly()){
```
Comment (chornesays) [inline Simulator/Simulator.cs:479]: Does this fire infinitely if someone deletes a tracker?

File: Simulator/Simulator.cs [hunk 3]
```diff
@@ -16,6 +20,8 @@ public class Simulator : MonoBehaviour
     
     private FaceRecording _faceRecording;
 
+    [SerializeField]
+    private Transform filterObject;
```
Comment (chornesays) [inline Simulator/Simulator.cs:24]: i missed this last time. private variables hsould have a _ prefix

**Review by chornesays — APPROVED** [review]
Generally approved. A few concerns around what happens if an artist deletes one of the trackers.

File: Simulator/Simulator.cs (hunk 3 above)
Comment (gazuntype) [reply Simulator/Simulator.cs:24]: ah noted.

File: Simulator/Simulator.cs (hunk 2 above)
Comment (gazuntype) [reply Simulator/Simulator.cs:479]: yup. until they fix the object. this ensures that the user knows something is definitely broken.

File: Simulator/Simulator.cs (hunk 1 above)
Comment (gazuntype) [reply Simulator/Simulator.cs:175]: if the trackers are null, nothing would be updated.

**Pull request merged**